sfcasts/autoconfiguration.md

```diff
@@ -1,6 +1,49 @@
+
+Down here, let's update this method. This should say `$this->issLocationPool`, `$this->client`, and since we can call it directly from the anonymous function, we don't need `use ($client)` anymore. Okay, at the browser, *refresh* to see... an *error*.
+
+`Variable "issData" does not exist.`
+
+Hm... We removed that from the controller. Open `/templates/main/homepage.html.twig` and, below, let's add our custom Twig function. Say `{% set issData = get_iss_location_data() %}`. If we refresh the page again... our custom function is *working*. But wait... how does Twig know to use this class? We didn't add any configuration from the Twig extension. Is it looking at the `/src/Twig/` directory? *Not exactly*. We could rename this directory and it would still work.
+
+The reason this works is thanks to the `autoconfigure: true` option in `/config/services.yaml`. Symfony *automatically* configures all of our services, like this Twig extension or even the `ShipReportCcommand` we created earlier. When that option is enabled, it basically tells Symfony:
+
+`Hey! Please look at the base class or interface
+of each service. If you're not an event subscriber
+or any other class that hooks into a part of
+Symfony, please go ahead and integrate that
+service into that system.`
+
+Yep! Symfony sees that our class extends a base command class and *knows* that it's a command that should be integrated into the system. In our extension's case, it *extends* `AbstractExtension`, so Symfony knows it should be integrated into the Twig system. I *love* this! It means that the only thing we need to care about is creating a PHP class that extends a certain class or implements a specific interface. The documentation will help you navigate this, and autoconfiguration will do the rest.
+
+Internally, autoconfiguration just adds a special tag for our services, like `console.command`, that helps the system notice it. But *other* times, it works via an *attribute*, like the command. In both cases, we create a class as the interface or attribute and *bam* - Symfony understands what you're doing and integrates it.
+
+All right, in `homepage.html.twig`, copy this HTML code, delete it, and open `base.html.twig`. Down here, below our logo, *paste*. Okay, let's simplify this a little. Create a new `<div>` above this and, inside, say `ISS Location`. Then, in parentheses, say `{{ issData.visibility }}`. We'll also give our `<div>` a `title` and set that to this `Updated At` line from below. Now we can clean up our code. We don't need this `<h2>`, we moved `Updated At` so that's no longer necessary here, and we can also get rid of `Visibility`. *Much* better! Over at our browser, refresh and... the ISS information is in a new position in the header. And if we open a different page, like one of our ship pages, the ISS location info is there too.
```

Comment:
```suggestion
All right, in `homepage.html.twig`, copy this HTML code, delete it, and open `base.html.twig`. Down here, below our logo, *paste*. Okay, let's simplify this a little in order to make it more compact. Create a new `<div>` above this and, inside, say `ISS Location`. Then, in parentheses, say `{{ issData.visibility }}`. We'll also give our `<div>` a `title` and set that to this `Updated At` line from below. Now we can clean up our code. We don't need this `<h2>`, we moved `Updated At` so that's no longer necessary here, and we can also get rid of `Visibility`. *Much* better! Over at our browser, refresh and... the ISS information is in a new position in the header. And if we open a different page, like one of our ship pages, the ISS location info is there too.
```